packages/components/src/tab-panel/index.js

```diff
@@ -61,6 +61,11 @@ export default function TabPanel( {
 		}
 	}, [ tabs ] );
 
+	const childrenResult = useMemo(
+		() => selectedTab && children( selectedTab ),
+		[ selectedTab, children ]
+	);
```

Comment:
Same here, Maybe I'm wrong but I don't think memoizing render functions is a good idea. It returns just an element and React already does the comparison of the rendered elements anyway.